java/mykola-melnyk/src/com/zoolatech/lecture4/tasks/_2/BinarySearchTree.java

```diff
@@ -0,0 +1,44 @@
+package com.zoolatech.lecture4.tasks._2;
+
+public class BinarySearchTree {
+    private Node rootNode;
+    public BinarySearchTree () {
+        rootNode = null;
+    }
+
+    private static class Node {
+        private final int value;
+        private Node leftChild;
+        private Node rightChild;
+        public Node (int value) {
+            this.value = value;
+            leftChild = null;
+            rightChild = null;
+        }
+    }
+
+    private void addRecursive (Node current, int value) {
+        if (current.value < value) {
+            if (current.rightChild == null) {
+                current.rightChild = new Node(value);
+            } else {
+                addRecursive(current.rightChild, value);
+            }
+        } else if (current.value > value) {
+            if (current.leftChild == null) {
+                current.leftChild = new Node(value);
+            } else {
+                addRecursive(current.leftChild, value);
+            }
+        }
+    }
+
+    public void add(int value) {
```

Comment:
methods should be defined in the usage order: method `add` uses the method `addRecursive`, so it's better to place `addRecursive` after `add`

Reply:
Done